juju/api.go

```diff
@@ -160,16 +160,18 @@ func NewAPIConnection(args NewAPIConnectionParams) (_ api.Connection, err error)
 		if accountDetails != nil {
 			accountType = accountDetails.Type
 		}
-		if ok && !user.IsLocal() && apiInfo.Tag == nil && accountType != jujuclient.OAuth2DeviceFlowAccountDetailsType {
-			// We used macaroon auth to login; save the username
-			// that we've logged in as.
-			accountDetails = &jujuclient.AccountDetails{
-				Type:            jujuclient.UserPassAccountDetailsType,
-				User:            user.Id(),
-				LastKnownAccess: st.ControllerAccess(),
+		if accountType == "" || accountType == jujuclient.UserPassAccountDetailsType {
```

Comment:
We can look to make this cleaner as well; maybe it can call the same function

Reply:
done.. extracted account detail processing into its own function